packages/client/src/pages/leaderboard/leaderboard.tsx

```diff
@@ -1,37 +1,72 @@
-
-import { useAppSelector } from '../../store/hooks'
-import { getUser } from '../../store/selectors'
-import { getPlayersStats } from '../../store/selectors'
+import { useAppDispatch, useAppSelector } from '../../store/hooks'
+import {
+  getLeaderboardState,
+  getUser,
+  isFirstPage,
+  isLastPage,
+  isLeaderboardLoading,
+} from '../../store/selectors'
 import { NavigationBar } from '../../components/navigationBar/navigationBar'
 import { LeaderboardTable } from '../../components/leaderboardTable/leaderboardTable'
 import heroImg from '../../assets/images/hero.png'
-
 import baseStyles from '../../app/app.module.css'
 import styles from './leaderboard.module.css'
+import { Button } from '../../components/button/button'
+import { nextPage, previousPage } from '../../store/leadeboard/leaderboardSlice'
+import { leaderboardThunks } from '../../store/leadeboard/leaderboardThunks'
+import { useEffect } from 'react'
+import { ActivityIndicator } from '../../components/activityIndicator/activityIndicator'
 
 export const Leaderboard = () => {
   const userState = useAppSelector(getUser)
-  const playersStatsState = useAppSelector(getPlayersStats)
+  const leaderboardState = useAppSelector(getLeaderboardState)
+  const isLoading = useAppSelector(isLeaderboardLoading)
+  const isFirst = useAppSelector(isFirstPage)
+  const isLast = useAppSelector(isLastPage)
+  const dispatch = useAppDispatch()
 
-  const navBar = userState.isAuth ? <NavigationBar /> : null
+  useEffect(() => {
+    dispatch(leaderboardThunks.setLeadersStatsState())
+  }, [])
 
-  return (
-    <div className={styles.leaderboard}>
-      {navBar}
-      <div className={styles.wrapper}>
-        <div className={styles.tableCaption}>Таблица лидеров</div>
-        <LeaderboardTable playersStats={playersStatsState} />
+  const leaderboard = (
+    <div className={styles.container}>
+      <div className={styles.tableCaption}>Таблица лидеров</div>
+      <LeaderboardTable state={leaderboardState} />
+
+      <div className={styles.buttonWrapper}>
+        <Button
+          onClick={() => dispatch(previousPage())}
+          className={isFirst ? styles.dimmed : ''}>
+          &lt;{' '}
+        </Button>
+        <Button
+          onClick={() => dispatch(nextPage())}
+          className={isLast ? styles.dimmed : ''}>
+          &gt;
+        </Button>
         <Link className={baseStyles.linkButton} to="/">
           На главную
         </Link>
-        <img
-          className={styles.heroImg}
-          src={heroImg}
-          alt="hero-img"
-          draggable="false"
-        />
       </div>
+      <img
+        className={styles.heroImg}
+        src={heroImg}
+        alt="hero-img"
+        draggable="false"
+      />
+    </div>
+  )
+
+  const navBar = userState.isAuth ? <NavigationBar /> : null
+  const pageContent = isLoading ? <ActivityIndicator /> : leaderboard
+
+  return (
+    <div className={styles.leaderboard}>
+      {navBar}
+
+      {pageContent}
     </div>
```

Comment:
Кажется, что вынос верстки должен упростить понимание, но тут получается наоборот. Так лучше не делать, потому что когда видишь большой кусок верстки в компоненте, это автоматически воспринимается как его `return`. 

Если хочешь вынести какую-то верстку из `return`, это должен быть вынос в отдельный компонент, желательно ещё в отдельном файле.

Конкретно тут на самом деле не вижу больших проблем в том, чтобы всё это просто положить в `return`

return (
```suggestion
  return (
    <div className={styles.leaderboard}>
      {userState.isAuth ? <NavigationBar /> : null}
      {isLoading ? <ActivityIndicator /> : (
         <<верстка лидерборда>>
      )}
    </div>
```
)

Reply:
Исправил.